File: plugins/kraken_bundle/ios/Classes/KrakenBundlePlugin.m
#import "KrakenBundlePlugin.h"
#import "KrakenBundleManager.h"

@implementation KrakenBundlePlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"kraken_bundle"
            binaryMessenger:[registrar messenger]];
  KrakenBundlePlugin* instance = [[KrakenBundlePlugin alloc] init];
  [[KrakenBundleManager shareBundleManager] onAttach:instance channel:channel];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getBundleUrl" isEqualToString:call.method]) {
    result([[KrakenBundleManager shareBundleManager] getBundleUrl]);
  } else if([@"getZipBundleUrl" isEqualToString:call.method]) {
    result([[KrakenBundleManager shareBundleManager] getZipBundleUrl]);
  } else {
    result(FlutterMethodNotImplemented);
  }
}
@end

File: plugins/kraken_bundle/macos/Classes/KrakenBundlePlugin.swift
import Cocoa
import FlutterMacOS

public class KrakenBundlePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "kraken_bundle", binaryMessenger: registrar.messenger)
    let instance = KrakenBundlePlugin()
    KrakenBundleManager.shared.onAttach(krakenBundlePlugin: instance, channel: channel)
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getBundleUrl":
      result(KrakenBundleManager.shared.getBundleUrl())
    case "getZipBundleUrl":
      result(KrakenBundleManager.shared.getZipBundleUrl())
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
